easybuild/easyblocks/c/clang.py

```diff
@@ -407,21 +408,54 @@ def build_with_prev_stage(self, prev_obj, next_obj):
+        prev_obj_path = os.path.join(prev_obj, 'bin')
+        current_path = read_environment({'path': 'PATH'})
+        setvar('PATH', current_path['path'] + ":" + prev_obj_path)
+
+        # If building with rpath, create RPATH wrappers for the Clang compilers for stage 2 and 3
+        if build_option('rpath'):
+            from easybuild.toolchains.compiler.clang import Clang
+            my_clang_toolchain = Clang(name='Clang', version='1')
+            my_clang_toolchain.prepare_rpath_wrappers()
+            self.log.info("Prepared clang rpath wrappers")
+
+            # RPATH wrappers add -Wl,rpath arguments to all command lines, including when it is just compiling
+            # Clang by default warns about that, and then some configure tests use -Werror which turns those warnings
+            # into errors. As a result, those configure tests fail, even though the compiler supports the requested
+            # functionality (e.g. the test that checks if -fPIC is supported would fail, and it compiles without
+            # resulting in relocation errors).
+            # See https://github.com/easybuilders/easybuild-easyblocks/pull/2799#issuecomment-1270621100
+            # Here, we add -Wno-unused-command-line-argument to CXXFLAGS to avoid these warnings alltogether
+            cflags = os.getenv('CFLAGS')
+            cxxflags = os.getenv('CXXFLAGS')
+            setvar('CFLAGS', "%s %s" % (cflags, '-Wno-unused-command-line-argument'))
+            setvar('CXXFLAGS', "%s %s" % (cxxflags, '-Wno-unused-command-line-argument'))
 
+        # Configure.
         options = "-DCMAKE_INSTALL_PREFIX=%s " % self.installdir
-        options += "-DCMAKE_C_COMPILER='%s' " % CC
-        options += "-DCMAKE_CXX_COMPILER='%s' " % CXX
+        options += "-DCMAKE_C_COMPILER='clang' "
+        options += "-DCMAKE_CXX_COMPILER='clang++' "
         options += self.cfg['configopts']
-        options += "-DCMAKE_BUILD_TYPE=%s" % self.build_type
+        options += "-DCMAKE_BUILD_TYPE=%s " % self.build_type
+
+        # Cmake looks for llvm-link by default in the same directory as the compiler
+        # However, when compiling with rpath, the clang 'compiler' is not actually the compiler, but the wrapper
+        # Clearly, the wrapper directory won't llvm-link. Thus, we pass the linker to be used by full path.
+        # See https://github.com/easybuilders/easybuild-easyblocks/pull/2799#issuecomment-1275916186
+        if build_option('rpath'):
+            llvm_link = which('llvm-link')
+            options += "-DLIBOMPTARGET_NVPTX_BC_LINKER=%s" % llvm_link
 
         self.log.info("Configuring")
         run_cmd("cmake %s %s" % (options, self.llvm_src_dir), log_all=True)
 
         self.log.info("Building")
-        run_cmd("make %s" % self.make_parallel_opts, log_all=True)
+        run_cmd("make %s VERBOSE=1" % self.make_parallel_opts, log_all=True)
+
+        # Restore environment (specifically: PATH)
+        restore_env(orig_env)
```

Comment:
We should just restore `$PATH` here, there's no need to fiddle with the rest of the environment at all.